web3_proxy/src/stats/mod.rs

```diff
@@ -224,26 +232,14 @@ impl BufferedRpcQueryStats {
         self.latest_balance = stat.latest_balance;
     }
 
-    /// Check a user's balance and possibly downgrade him in the cache
-    async fn downgrade_user(self) {}
-
-    // TODO: take a db transaction instead so that we can batch?
-    async fn save_db(
-        self,
+    async fn _save_db_stats(
+        &self,
         chain_id: u64,
         db_conn: &DatabaseConnection,
-        key: RpcQueryKey,
+        key: &RpcQueryKey,
         rpc_secret_key_cache: RpcSecretKeyCache,
         user_balance_cache: UserBalanceCache,
     ) -> Web3ProxyResult<()> {
-        if key.response_timestamp == 0 {
```

Comment:
This is a safety check, so I think we want to keep this. Though I suppose it could be a debug_assert

Reply:
just to confirm, you want to put this into _save_db_stats()?

Right now, my code shows me this: 
```
async fn save_db(
        self,
        chain_id: u64,
        db_conn: &DatabaseConnection,
        key: RpcQueryKey,
        rpc_secret_key_cache: &RpcSecretKeyCache,
        user_balance_cache: &UserBalanceCache,
    ) -> Web3ProxyResult<()> {
        if key.response_timestamp == 0 {
            return Err(Web3ProxyError::Anyhow(anyhow!(
                "no response_timestamp! This is a bug! {:?} {:?}",
                key,
                self
            )));
        }

        // First of all, save the statistics to the database:
        self._save_db_stats(
            chain_id,
            db_conn,
            &key,
            rpc_secret_key_cache,
            user_balance_cache,
        )
        .await?;
        ```